src/main.cpp

```diff
@@ -64,1577 +66,18 @@ using namespace kodi::tools;
-      res |= (1U << adp2ips[i]);
-  return res;
-}
-
-STREAM_CRYPTO_KEY_SYSTEM Session::GetCryptoKeySystem() const
-{
-  if (m_kodiProps.m_licenseType == "com.widevine.alpha")
-    return STREAM_CRYPTO_KEY_SYSTEM_WIDEVINE;
-#if STREAMCRYPTO_VERSION_LEVEL >= 1
-  else if (m_kodiProps.m_licenseType == "com.huawei.wiseplay")
-    return STREAM_CRYPTO_KEY_SYSTEM_WISEPLAY;
-#endif
-  else if (m_kodiProps.m_licenseType == "com.microsoft.playready")
-    return STREAM_CRYPTO_KEY_SYSTEM_PLAYREADY;
-  else
-    return STREAM_CRYPTO_KEY_SYSTEM_NONE;
-}
-
-int Session::GetChapter() const
-{
-  if (adaptiveTree_)
-  {
-    std::vector<adaptive::AdaptiveTree::Period*>::const_iterator res =
-        std::find(adaptiveTree_->periods_.cbegin(), adaptiveTree_->periods_.cend(),
-                  adaptiveTree_->current_period_);
-    if (res != adaptiveTree_->periods_.cend())
-      return (res - adaptiveTree_->periods_.cbegin()) + 1;
-  }
-  return -1;
-}
-
-int Session::GetChapterCount() const
-{
-  if (adaptiveTree_)
-    return adaptiveTree_->periods_.size() > 1 ? adaptiveTree_->periods_.size() : 0;
-  return 0;
-}
-
-const char* Session::GetChapterName(int ch) const
-{
-  --ch;
-  if (ch >= 0 && ch < static_cast<int>(adaptiveTree_->periods_.size()))
-    return adaptiveTree_->periods_[ch]->id_.c_str();
-  return "[Unknown]";
-}
-
-int64_t Session::GetChapterPos(int ch) const
-{
-  int64_t sum(0);
-  --ch;
-
-  for (; ch; --ch)
-    sum += (adaptiveTree_->periods_[ch - 1]->duration_ * STREAM_TIME_BASE) /
-           adaptiveTree_->periods_[ch - 1]->timescale_;
-  return sum / STREAM_TIME_BASE;
-}
-
-uint64_t Session::GetChapterStartTime() const
-{
-  uint64_t start_time = 0;
-  for (adaptive::AdaptiveTree::Period* p : adaptiveTree_->periods_)
-    if (p == adaptiveTree_->current_period_)
-      break;
-    else
-      start_time += (p->duration_ * STREAM_TIME_BASE) / p->timescale_;
-  return start_time;
-}
-
-int Session::GetPeriodId() const
-{
-  if (adaptiveTree_)
-  {
-    if (IsLive())
-      return adaptiveTree_->current_period_->sequence_ == adaptiveTree_->initial_sequence_
-                 ? 1
-                 : adaptiveTree_->current_period_->sequence_ + 1;
-    else
-      return GetChapter();
-  }
-  return -1;
-}
 
-bool Session::SeekChapter(int ch)
-{
-  if (adaptiveTree_->next_period_)
-    return true;
+/*******************************************************
+Main class Session
+********************************************************/
 
-  --ch;
-  if (ch >= 0 && ch < static_cast<int>(adaptiveTree_->periods_.size()) &&
-      adaptiveTree_->periods_[ch] != adaptiveTree_->current_period_)
-  {
-    adaptiveTree_->next_period_ = adaptiveTree_->periods_[ch];
-    for (auto& stream : m_streams)
-    {
-      if (stream->GetReader())
-        stream->GetReader()->Reset(true);
-    }
-    return true;
-  }
 
-  return false;
-}
 
 /***************************  Interface *********************************/
```

Comment:
i know that there are a lot of changes but i think is appropriate
move to the header the definition of classes
And also the forward `class CInputStreamAdaptive;` of line 84 should be not more needed